### PR TITLE
[SPARK-45502][BUILD] Upgrade Kafka to 3.6.1

### PR DESCRIPTION
**Pull request by dengziming**
### What changes were proposed in this pull request?
Upgrade Apache Kafka from 3.4.1 to 3.6.1

### Why are the changes needed?

- https://downloads.apache.org/kafka/3.6.1/RELEASE_NOTES.html
- https://downloads.apache.org/kafka/3.6.0/RELEASE_NOTES.html
- https://downloads.apache.org/kafka/3.5.0/RELEASE_NOTES.html

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
GitHub CI.

### Was this patch authored or co-authored using generative AI tooling?
No

Closes #43348 from dengziming/kafka-3.6.0.

Authored-by: dengziming <dengziming1993@gmail.com>
Signed-off-by: Dongjoon Hyun <dhyun@apple.com>

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you, @dengziming . Could you make the CI happy, please?

**Comment by dengziming** [comment]
@dongjoon-hyun It seems `SparkSessionE2ESuite` failed once and it's not related to this PR, right?

File: connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala [hunk 1]
```diff
@@ -2691,7 +2691,9 @@ class KafkaSourceStressSuite extends KafkaSourceTest {
     start + Random.nextInt(start + end - 1)
   }
 
-  test("stress test with multiple topics and partitions")  {
+  override val brokerProps = Map("auto.create.topics.enable" -> "false")
```
Comment (dengziming) [inline connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala:2694]: Add this property to reduce flakiness.

Comment (dongjoon-hyun) [reply connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala:2694]: Thanks!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @dengziming .
Merged to master for Apache Spark 4.0.0.

**Pull request closed without merging**